contrib/lkt/language/parser.py

```diff
@@ -2208,6 +2208,41 @@ def expected_type_predicate(expected_type=T.TypeDecl.entity):
     def invalid_expected_type_error_name():
         return S("an array")
 
+    @langkit_property()
+    def expr_context_free_type():
+        """
+        Infer type from array elements context_free_type.
+        """
+        # Get the context_free_type of the array elements if any
+        types = Var(
+            Entity.exprs.filter(
+                # Filter out Lit expressions to avoid infinite recursion
+                lambda e: Not(
+                    Or(e.is_a(T.NumLit),
+                       e.is_a(T.CharLit),
+                       e.is_a(T.StringLit))
```

Comment:
`.is_a()` accepts multiple type arguments, so this can be simplified to: `Not(e.is_a(T.NumLite, T.CharLit, T.StringLit))`